### PR TITLE
Remove plugin specific keys

### PR DESCRIPTION
**Pull request by huntc**
Mainly removed the plugin specific keys and employed the bintray plugin logic.

The project has also been greatly enhanced in order to support scalariform, the release plugin and others. A little test sub project is provided to aid development also.

Package creation, license checking and other things that the bintray plugin does has now also been incorporated.

Finally ConductR compatibility version meta data has also been incorporated.

OK, tons of things have changed. :-P

Related issue and motivation for this change: https://github.com/typesafehub/conductr-cli/issues/72

### CODE REVIEW

**Comment by huntc** [comment]
@eed3si9n Still a WIP, so not looking for a merge, but interested in your feedback.

File: src/main/scala/com/typesafe/sbt/bintraybundle/BintrayBundle.scala [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        val packageName = (bintrayPackage in config).value
+        val repo = Bintray.cachedRepo(
+          (bintrayEnsureCredentials in config).value,
+          (bintrayOrganization in config).value,
+          (bintrayRepository in config).value
+        )
+        val bundleDist = (dist in config).value
+        val version = bundleDist.getName.split("-").toList.takeRight(2).mkString("-").replaceAll("""\.zip$""", "")
+        val bundleExt = ".zip"
+        val path = s"/${repo.owner}/$packageName/$version/$packageName-$version$bundleExt"
```
Comment (fsat) [inline src/main/scala/com/typesafe/sbt/bintraybundle/BintrayBundle.scala]: @huntc - looking back at the urn convention for bundle in https://github.com/typesafehub/conductr-cli/issues/72 - is `${repo.owner}` actually `repository` not `organisation`?

**Comment by fsat** [comment]
Lgtm, at least from what the ConductR CLI expects.

**Comment by huntc** [comment]
Have had to re-declare private functions of the bintray plugin - see https://github.com/softprops/bintray-sbt/issues/76

**Pull request merged**